### PR TITLE
Added recorder to track last request and response (issue #47).

### PR DESCRIPTION
**Pull request by piotrooo**

### CODE REVIEW

File: tests/fXmlRpc/Transport/RecorderTest.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        $this->transportOk();
+        $this->client->call('TestMethod', ['param1', 2, ['param3' => true]]);
+
+        $lastResponse = $this->recorder->getLastResponse();
+
+        $this->assertXmlStringEqualsXmlString($this->expectedResponse, $lastResponse);
+    }
+
+    public function testReturnXmlForRequestAndNullForResponseWhenTransportThrowsException()
```
Comment (lstrojny) [inline tests/fXmlRpc/Transport/RecorderTest.php]: We could think about recording the exception as well. Basically: `getLastException() : null|Exception`

File: tests/fXmlRpc/Transport/RecorderTest.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            $this->transportFail();
+            $this->client->call('TestMethod', ['param1', 2, ['param3' => true]]);
+        } catch (\Exception $e) {
+        }
+
+        $lastRequest = $this->recorder->getLastRequest();
+        $lastResponse = $this->recorder->getLastResponse();
+
+        $this->assertNotNull($lastRequest);
+        $this->assertNull($lastResponse);
```
Comment (lstrojny) [inline tests/fXmlRpc/Transport/RecorderTest.php]: There is one behaviour left that is uncovered and potentially confusing:

 - First request: everything works and request and response are populated
 - Second request: exception is thrown, request is populated but response is still the one from the first request which I would say is confusing

In order to fix that case we should cover it and reset the last response before we call the wrapped transport.

File: tests/fXmlRpc/Transport/RecorderTest.php (hunk 1 above)
Comment (piotrooo) [inline tests/fXmlRpc/Transport/RecorderTest.php]: Seems logic for me. In this case I proposes set:
request: xml
response: null
exception: throwed exception

Comment (lstrojny) [inline tests/fXmlRpc/Transport/RecorderTest.php]: Makes sense

**Comment by lstrojny** [comment]
Manually merged. Would you mind open another PR to update the README?

**Pull request closed without merging**